source/game/system/map/MapdataCheckPoint.cc

```diff
@@ -95,15 +95,120 @@ void MapdataCheckPoint::initCheckpointLinks(MapdataCheckPointAccessor &accessor,
     }
 }
 
+/// @addr{0x80510D7C}
+/// @see MapdataCheckPoint::checkSectorAndDistanceRatio
+/// @rename
+MapdataCheckPoint::SectorOccupancy MapdataCheckPoint::checkSectorAndDistanceRatio(
+        const EGG::Vector3f &pos, f32 &completion) const {
+    EGG::Vector2f p1 = m_right;
+    p1.y = pos.z - p1.y;
+    p1.x = pos.x - p1.x;
+
+    for (auto &next : m_nextPoints) {
+        EGG::Vector2f p0 = next.checkpoint->m_left;
+        p0.y = pos.z - p0.y;
+        p0.x = pos.x - p0.x;
+        MapdataCheckPoint::SectorOccupancy result =
+                checkSectorAndDistanceRatio(next, p0, p1, completion);
+
+        if (result != SectorOccupancy::OutsideSector) {
+            return result;
+        }
+    }
+
+    return SectorOccupancy::OutsideSector;
+}
+
+void MapdataCheckPoint::setSearched() {
+    m_searched = true;
+}
+
+void MapdataCheckPoint::clearSearched() {
+    m_searched = false;
+}
+
+bool MapdataCheckPoint::searched() const {
+    return m_searched;
+}
+
 s8 MapdataCheckPoint::checkArea() const {
     return m_checkArea;
 }
 
+bool MapdataCheckPoint::isNormalCheckpoint() const {
+    return static_cast<MapdataCheckPoint::CheckArea>(m_checkArea) ==
+            MapdataCheckPoint::CheckArea::NORMAL_CHECKPOINT;
+}
+
+bool MapdataCheckPoint::isFinishLine() const {
+    return static_cast<MapdataCheckPoint::CheckArea>(m_checkArea) ==
+            MapdataCheckPoint::CheckArea::FINISH_LINE;
+}
```

Comment:
```suggestion
bool MapdataCheckPoint::isNormalCheckpoint() const {
    return static_cast<CheckArea>(m_checkArea) == CheckArea::NORMAL_CHECKPOINT;
}

bool MapdataCheckPoint::isFinishLine() const {
    return static_cast<CheckArea>(m_checkArea) == CheckArea::FINISH_LINE;
}
```
We're scoped inside `MapdataCheckPoint`.